### PR TITLE
[3rdparty] Bump doctrine/dbal to 2.12.0

### PR DESCRIPTION
**Pull request by rullzer**
Needed for php8 support

Requires:
- [x] https://github.com/nextcloud/3rdparty/pull/517
- [x] rebase

### CODE REVIEW

**Comment by rullzer** [comment]
Needed for php8 stuff.
I don't think we can get rid of the psalm warnings here easily.

**Comment by MorrisJobke** [comment]
> I don't think we can get rid of the psalm warnings here easily.

Just update the baseline and we can move forward then.

**Review by nickvergessen — APPROVED** [review]

**Comment by MorrisJobke** [comment]
> Just update the baseline and we can move forward then.

Please update the baseline and then we can merge this :)

File: build/psalm-baseline.xml [hunk 1]
```diff
@@ -1,52 +1,31 @@
...
-    </MoreSpecificImplementedParamType>
-  </file>
-  <file src="3rdparty/sabre/dav/lib/CardDAV/AddressBookHome.php">
-    <InvalidPropertyAssignmentValue occurrences="1">
-      <code>$principalUri</code>
-    </InvalidPropertyAssignmentValue>
-    <MoreSpecificImplementedParamType occurrences="1">
-      <code>$data</code>
-    </MoreSpecificImplementedParamType>
+  <file src="/home/roeland/nc/3rdparty/composer/autoload_real.php">
```
Comment (MorrisJobke) [inline build/psalm-baseline.xml:3]: looks like your version includes absolute paths and not relative ones 😭

Comment (ChristophWurst) [reply build/psalm-baseline.xml:3]: we can just all rename our user to `roeland` and it will work fine

Comment (rullzer) [reply build/psalm-baseline.xml:3]: > looks like your version includes absolute paths and not relative ones sob

well then you run the update.. I'm not sure what I can do it is the default psalm from composer install

File: build/psalm-baseline.xml [hunk 2]
```diff
@@ -3554,6 +3562,9 @@
     <InvalidScalarArgument occurrences="1">
       <code>$e-&gt;getCode()</code>
     </InvalidScalarArgument>
+    <LessSpecificImplementedReturnType occurrences="2">
+      <code>Doctrine\DBAL\Driver\Statement</code>
```
Comment (MorrisJobke) [inline build/psalm-baseline.xml:3566]: That changes from `Doctrine\DBAL\Statement` upstream to `Doctrine\DBAL\Driver\Statement` on our side and may be a bigger problem.

**Review by MorrisJobke — APPROVED** [review]
Fine with me now 👍

**Comment by rullzer** [comment]
Mmmm we still need a psalm baseline update. Lets do that separatly then. As to not delay this (as php8 stuff also depends on it).

**Comment by MorrisJobke** [comment]
Psalm fix is in #23834

**Pull request merged**